File: 백준/Bronze/20124. 모르고리즘 회장님 추천 받습니다/모르고리즘 회장님 추천 받습니다.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        List<Person> lst = new ArrayList<>();
        
        for (int i = 0; i < N; i++) {
            String[] input = br.readLine().split(" ");
            String name = input[0];
            int score = Integer.parseInt(input[1]);
            lst.add(new Person(name, score));
        }

        Collections.sort(lst, (p1, p2) -> {
            if (p2.score != p1.score) {
                return Integer.compare(p2.score, p1.score);
            } else {
                return p1.name.compareTo(p2.name);
            }
        });
        System.out.println(lst.get(0).name);
    }

    static class Person {
        String name;
        int score;
        Person(String name, int score) {
            this.name = name;
            this.score = score;
        }
    }
}

File: 백준/Bronze/20124. 모르고리즘 회장님 추천 받습니다/모르고리즘 회장님 추천 받습니다.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    N = int(input())
    lst = []
    for _ in range(N):
        n, s = input().split()
        lst.append((n, int(s)))
    lst.sort(key=lambda x: (-x[1], x[0]))
    print(lst[0][0])


if __name__ == '__main__':
    main()